instructor_notes/11_git-init-server.md

```diff
@@ -0,0 +1,116 @@
+# Instructor Notes: Initializing a git repository on RStudio Server
+
+This document provides instructions for creating a git repository in an existing directory on the Data Lab's RStudio Server, as well as setting up associated git credentials.
+
+## Learning goals
+
+At the end of this activity, workshop attendees will have:
+
+* Learned about git Personal Access Tokens and credentials
+* Learned basic git configuration using R tools
+* Used basic git CLI commands
+
+## Part 1: Set up git credentials
+
+### Set up the `~/.gitconfig` file
+
+This can either be done directly in the editor by opening `~/.gitconfig`, or with `usethis` which may be safer for ensuring correct spacing:
+
+In the end, the `~/.gitconfig` file should look like this (where the `init` block is optional but recommended):
+
+```yml
+[user]
+    email = Participant Email
+    name = Participant Name
+[credential]
+    helper = cache --timeout=43200
+[init]
+    defaultBranch = "main"
+```
+
+The `usethis` code to add this content to `~/.gitconfig` is below.
+If `usethis` is used, instruct participants to view the `~/.gitconfig` file after each step to see how the file has been modified.
+Note that `usethis::use_git_config()` will not modify (or delete!) other fields besides those provided to the function.
+
+```r
+# Step 1: Create config file with user and email
+usethis::use_git_config(
+    user.name = "Participant Name",
+    user.email = "Participant Email",
+    # Optional setting to ensure the config is global for the user
+    scope = "user",
+    # Optional setting to use `main` as default branch when creating a new repo
+    init.defaultBranch = "main"
+)
+# Step 2: Set up credential helper to be able to temporarily store PAT on the server
+# Note that this could be combined with step 1 as an additional argument
+usethis::use_git_config(
+    credential.helper = "cache --timeout=43200" # 12 hours
+)
+```
+
+### Add your Personal Access Token
+
+* Instruct participants to navigate to their GitHub.com account's Developer settings and create a classic token with the `repo` scope that lasts for 7 days.
```

Comment:
More detail needed here, I think.  Full "path" to the page Settings> Developer Settings > Personal Access Tokens
Maybe say a bit more about permissions:`repo` is most of what they need, but I would guess that `user` permissions are also potentially useful, and maybe `gist`. They will also need to fill out the "Note" field.

Reply:
When I tested this out with just add/commit/push/pull, `repo` was enough to get the job done. Where do you see `user` coming into play - as something they'd potentially need for the workshop, or as something to introduce them to for their future PAT needs?

Reply:
I mostly just wanted to make sure we talk about them a bit.